### ISSUE TITLE
Enable attach and display logs on 3rd level by default

### ISSUE THREAD

**Issue by kogeletey** [pipelight/pipelight#2]
Often a situation arises when you need know what output your instrument produces.
This is especially useful if we use it with git hooks.
In connection with the above I suggest displaying logs on 3rd level by default, for example [moby/buildkit](https://github.com/moby/buildkit) does so, and if it don't need the disable them using `--no-attach`.

**Comment by pipelight** [comment]
So! Let me rephrase in my own words.
If I understand, you want the pipeline to be attached when run through git-hooks.

So that, whenever something unexpected happens on a `git push` for example:
- You can see the displayed logs.
- The git operation is halted.

I like pipelines detached by default, because long running jobs can run without stopping or halting git commands.
But if this behavior is to unconventional and harm the users software quality, i will change asap default to attached.

I came to an option,
that could circumvent the issue.

A global and a per pipeline flag to override the default behavior.

```ts
const config: Config = {
  pipelines: [{
     steps:[],
     attach: true
  }],
  attach: true
};

```

In the case where attached becomes the default behavior:

```ts
const config: Config = {
  pipelines: [{
     steps:[],
     detach: true
  }],
  detach: true
};

```

**Comment by kogeletey** [comment]
The last option really looks good only if you also add the log settings.

I suggest some kind of option.

```typescript
// Yes, java bit me
type additionOfConfigOrPipelines = {
  // true - attach, false - detach,failure - when an error occurs
  attach: boolean|"failure"
  // -v = 1,-vv = 2,-vvv=3,-vvv=4
  logLevel: 1|2|3|4
}
```

If it is possible to do this, I will be very grateful and it will solve a lot of problems.

**Comment by pipelight** [comment]
Nice! Let's do it!

I'll need some time!

I'll come back in 24h with something!

**Comment by kogeletey** [comment]
Thank you very much, I hope this will improve the product and make it even more attractive.
I don’t know about others, but it seems logical to me to use `attach: false` and not add another option in the configuration.

**Comment by pipelight** [comment]
It's here for you to test! On the feature/detach_flag branch!

**Comment by kogeletey** [comment]
This is the fastest support I've seen so far!
After looking at the code, I may have missed it, but will it be possible to change logLevel globally?

**Comment by pipelight** [comment]
I am on it!

**Comment by kogeletey** [comment]
Thanks for your help and implementation

**Comment by kogeletey** [comment]
I checked and so far neither global config nor local pipelines options are working
```typescript
const config: Config = {
  pipelines: [
    lint,
  ],
  options: {
    attach: true,
    log_level: "info",
  },
}
```

Here is my configuration, and the task executes, but does not display in the terminal

**Comment by kogeletey** [comment]
The settings do now work for git-hooks, but still do not work for the run command.

**Comment by pipelight** [comment]
Ah! I am relieved it finally works!

This was intended to be as is:
I chose that `pipelight run`  will not be overridden by those settings because they can be set on the command line.
only `pipelight trigger` is impacted because it is a mess to dig inside the git hook and modify it manually.

**Comment by pipelight** [comment]
Ok! I'll modify the `run` command behavior

**Comment by pipelight** [comment]
Ah yes!

For the "info" log level, pipelight prints stdout on succes and stderr on fail.

Linters tend to avoid standards and print to both outputs.

I recommand setting the log level to "trace"  to display stdout and stderr when linters are involved.

**Comment by pipelight** [comment]
So! I don't know if this is easy for you to try it, but I made the changes available  into the dev branch!
The configuration settings now inpact the `pipelight run` command! and the arguments of the command line can overridde these settings.

As I may not have the time needed for deeper code review today, the changes will only be merged to master tomorrow.

**Comment by kogeletey** [comment]
Thank you for quickly introducing new features.
LogLevel doesn't work yet, but `attach` is ok.

**Comment by pipelight** [comment]
I needed to better understand the clap.rs crate. All done and pushed to master under v0.7.15

**Comment by kogeletey** [comment]
Yes, I checked everything, it really works great.
Thank you very much again for such a prompt solution to the problem.
